trace/metric.go

```diff
@@ -0,0 +1,70 @@
+package trace
+
+import (
+	"context"
+
+	"github.com/opentracing/opentracing-go"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/uber/jaeger-client-go"
+)
+
+// Counter is a wrapper of a prometheus.Counter.
+type Counter struct {
+	prometheus.Counter
+}
+
+// Add adds the given value to the counter. If there is a span associated with a context ctx the method
+// replaces the currently saved exemplar (if any) with a new one, created from the provided value.
+func (c *Counter) Add(ctx context.Context, count float64) {
+	spanFromCtx := opentracing.SpanFromContext(ctx)
+	if spanFromCtx != nil {
+		if sctx, ok := spanFromCtx.Context().(jaeger.SpanContext); ok {
+			c.Counter.(prometheus.ExemplarAdder).AddWithExemplar(
```

Comment:
Sorry, I'm missing the point about why do we need to make type assertions. If `Counter` is a `prometheus.ExemplarAdder`, then why can't we make this field a `prometheus.ExemplarAdder`,?

Reply:
A `Counter` is a [`prometheus.Counter` interface](https://pkg.go.dev/github.com/prometheus/client_golang/prometheus@v1.11.0#Counter).
Theoretically there could be a bespoke counter implementing `prometheus.Counter` interface but not `prometheus.ExemplarAdder` interface, that is why we need a type assertion here.
Or am I missing your point here?

Reply:
But if we talk about evolutions and that the `Counter` may be of another type later on, then this code will panic ([example](https://go.dev/play/p/a9T02uuxm-b)). How are we going to handle such a case?